### PR TITLE
[New] Add clarity-icons@0.10.2 w/ npm auto-update

### LINKED ISSUE #11600: [Request] Add clarity
**Library name:** clarity
**Git repository url**: https://github.com/vmware/clarity
**npm package name or url** (if there is one): https://www.npmjs.com/package/clarity-icons
**License (List them all if it's multiple):** MIT
**Official homepage:** http://clarity.design/
**Wanna say something? Leave message here:** 
Would love to have the clarity-ui css framework available on cdnjs

=====================

Notes from cdnjs maintainer:
Please read the [README.md](https://github.com/cdnjs/cdnjs#cdnjs-library-repository) and [CONTRIBUTING.md](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md) document first.

We encourage you to add a library via sending pull request,
it'll be faster than just opening a request issue,
since there are tons of issues, please wait with patience,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sufuf3**
close #11600
Pull request for issue: #11600
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/vmware/clarity
 * Official website (optional, not the repository): https://vmware.github.io/clarity/
 * NPM package url (optional): https://www.npmjs.com/package/clarity-icons
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 103
   - Count of stars: 1744
   - Count of forks: 181
 * NPM download stats (optional):
   - Downloads in the last day: 1735
   - Downloads in the last week: 9594
   - Downloads in the last month: 37,944

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sufuf3** [comment]
Although the latest version is `v0.10.0-rc.1`, I think it not the stable version, but cdnjs can host, so I add `v0.9.16`.
Although GitHub versions more than npm, according to [Installing Clarity Icons](https://github.com/vmware/clarity#installing-clarity-icons) it said Include the `clarity-icons.min.css` and `clarity-icons.min.js` in HTML, and the files I can get from npm. (GitHub can't find the files)

**Comment by Shijir** [comment]
@sufuf3 Thanks a lot for this PR. We don't keep Clarity build files in our Github repo. If you need any help on Clarity Icons, feel free to ping me.

**Comment by sufuf3** [comment]
OK, Thanks that pay attention to this PR. :)

File: .gitignore [hunk 1]
```diff
@@ -114,6 +114,7 @@ node_modules/
 /ajax/libs/strophe.js/release-1.2.0
 /ajax/libs/raphael/semver/
 /ajax/libs/clappr/media_control_youtube/
+/ajax/libs/clarity-icons/0.7.5-dev
```
Comment (extend1994) [inline .gitignore]: Maybe we can use `*-dev*` in case of future appearance?

File: ajax/libs/clarity-icons/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "clarity-icons",
+  "version": "0.9.16",
+  "filename": "clarity-icons.min.js",
+  "description": "Custom Element Icons for Clarity",
+  "homepage": "https://vmware.github.io/clarity/",
+  "repository": {
+    "type": "git",
+    "url": "ssh://git@git.eng.vmware.com/clarity.git"
```
Comment (extend1994) [inline ajax/libs/clarity-icons/package.json]: Not sure if this matches the format...?

**Comment by extend1994** [comment]
Thanks! @Shijir

**Comment by extend1994** [comment]
We may also add clarity-ui 😃?

**Comment by sufuf3** [comment]
@extend1994 yes, I'll add it

File: ajax/libs/clarity-icons/package.json (hunk 2 above)
Comment (sufuf3) [inline ajax/libs/clarity-icons/package.json]: I'll update to `git+https://github.com/vmware/clarity.git`
Although https://github.com/vmware/clarity/blob/master/build/npm/clarity-icons.json#L10, the link can't work.

File: .gitignore (hunk 1 above)
Comment (sufuf3) [inline .gitignore]: Just follow https://github.com/cdnjs/cdnjs/commit/94ea88b85054a45ca6d5b1f6ffc190eea2cedc78
I'll update this like https://github.com/cdnjs/cdnjs/commit/e61dc9c9271fc82e5d7e5260cedbaaf4efbb01f0

**Comment by sufuf3** [comment]
@extend1994 done
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11
Please help me review this PR, Thank you.

**Comment by Shijir** [comment]
@PeterDaveHello Agree, it would be nice to include the lighter version of `clarity-icons.min.js`, which is `clarity-icons-lite.min.js`.

But for the `clarity-icons-element.js`, users don't need `cdn` for that as it's an internal module. User shouldn't access it.

**Comment by Shijir** [comment]
Also, I just noticed/realized that this PR is based on `clarity-icons@0.9.16`. But `clarity-icons@0.10.2` is already out. And the `clarity-icons-lite.min.js` doesn't exist in the v0.9.x version.

**Comment by sufuf3** [comment]
@Shijir Thank you.
@cdnjs/active-contributor @cdnjs/intern3 I updated the pattern and version
Please help me review this PR, Thank you.

**Comment by PeterDaveHello** [comment]
@sufuf3 so what's the reason why the problem caused? It's very strange that I trained you for a whole year but you still do things your own way, once question came, you don't discuss about the issue, but directly do something else, is that how you improve the prevent the same problem appears again?

**Comment by Shijir** [comment]
@sufuf3 It's getting there. But if we are including the lighter version of the library, we also need to include the add-on shapes such as `commerce-shapes.js`,`essential-shapes.min.js`, `media-shapes.min.js`, `travel-shapes.min.js`,  and `technology-shapes.min.js`

Basically, `clarity-icons.min.js` = `clarity-icons-lite.min.js` + `commerce-shapes.min.js` + `essential-shapes.min.js`+ `media-shapes.min.js`+ `travel-shapes.min.js` + `technology-shapes.min.js`

The following npm package structure might help you to find the shape files I mentioned above.
```
clarity-icons
├── clarity-icons-lite.min.js
├── clarity-icons.min.css
├── clarity-icons.min.js
├── ...
├── shapes
│   ├── commerce-shapes.min.js
│   ├── essential-shapes.min.js
│   ├── media-shapes.min.js
│   ├── social-shapes.min.js
│   ├── technology-shapes.min.js
│   ├── travel-shapes.min.js
│   └── ...
├── ...

```

So the final version of this PR would include the following files:

1. clarity-icons.min.css
2. clarity-icons.min.js

3. clarity-icons-lite.min.js
4. commerce-shapes.min.js
5. essential-shapes.min.js
6. media-shapes.min.js
7. social-shapes.min.js
8. technology-shapes.min.js
9. travel-shapes.min.js

**Comment by sufuf3** [comment]
@Shijir Thank you.
I'll update it. :)

**Comment by sufuf3** [comment]
@cdnjs/active-contributor @cdnjs/intern3 
Please help me review this PR, Thank you.

**Pull request merged**